JS/Lesson34/onder/my-app/src/components/NewRecipe/index.jsx

```diff
@@ -1,77 +1,98 @@
 import { useForm } from 'react-hook-form';
 import './styles.css';
 
-// const useMyHook = () => {
-//   const myData = {id: 'xxxx', name:'Data name'};
-//   const deleteMyData = () => {console.log('Deleting data...')};
+export const NewRecipe = () => {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      strMeal: '',
+      strDrinkAlternate: '',
+      strCategory: '',
+      strCategoryOther: '',
+      strInstructions: '', 
+      strIngredientsAndMeasurements: '',
+    },
+  });
 
-//   return [ myData, deleteMyData ];
-// }
+  const strCategory = watch('strCategory');
 
-export const NewRecipe = () => {
-  // const [data,  setData] = useMyHook();
+  const onSubmit = (data) => {
+   
+    const ingredientsAndMeasurements = data.strIngredientsAndMeasurements.split(',').map(item => item.trim());
```

Comment:
The onSubmit function should look like this in the end:
```
	const onSubmit = (data) => {
		const ingredientsAndMeasuresData = getIngredientsAndMeasurements(data);

// Filter data to remove `strIngredientsAndMeasurements` that we have in the form as it's not compatible with API data
		const filteredData = Object.keys(data)
			.filter((key) => key !== 'strIngredientsAndMeasurements')
			.reduce((obj, key) => {
				obj[key] = data[key];
				return obj;
			}, {});

		console.log({
			...filteredData,
			...ingredientsAndMeasuresData,
		});
	};
```

It uses this function (added right outside the component not to pollute the component code): 
```
function getIngredientsAndMeasurements(formData) {
  const ingredientsAndMeasurementsArray = formData.strIngredientsAndMeasurements
      .split(',')
      .map(item => item.trim().split('-').map(part => part.trim()));

  const ingredientsObj = {};
  const measurementsObj = {};

  ingredientsAndMeasurementsArray.forEach(([ingredient, measurement], index) => {
      ingredientsObj[`strIngredient${index + 1}`] = ingredient;
      measurementsObj[`strMeasure${index + 1}`] = measurement;
  });

  return { ...ingredientsObj, ...measurementsObj };
}
```

This function converts {..., "strIngredientsAndMeasurements": "Flour - 100gr, milk - 50ml"} to {..., "strIngredient1": "Flour", "strIngredient2": "milk", "strMeasure1": "100gr", "strMeasure2": "50ml"}.

In real life event more validation would be required but it's ok not to do it right now fort his assignment.